asdf/tests/helpers.py

```diff
@@ -27,7 +27,7 @@
 from asdf.block import Block
 from asdf.constants import YAML_TAG_PREFIX
 from asdf.exceptions import AsdfConversionWarning, AsdfDeprecationWarning
-from asdf.extension import default_extensions
+from asdf.extension._legacy import _default_extensions as default_extensions
```

Comment:
Since this is a module often accessed as if it were in the public API, I think it would be better to not recast the name as "public" and instead leave the name as is. There is only one usage on line 101, so it is a simple fix.

If the deprecation changes I suggested for `_legacy` and `asdf.extension.__init__` are employed I think this should be instead turned into `from asdf.extension import _legacy` and then use `_legacy.default_extensions` on line 101 instead.